File: tests/fixtures/grammar/constructs.rb
# Every control-flow construct in the Ruby grammar that can carry a
# decision. One method per construct.

def if_elsif(v)
  if v > 0
    1
  elsif v < 0
    -1
  else
    0
  end
end

def modifier_if(v)
  return 0 if v.nil?
  v
end

def unless_statement(v)
  unless v.nil?
    return v
  end
  0
end

def while_loop(n)
  while n > 0
    n -= 1
  end
  n
end

def until_loop(n)
  until n <= 0
    n -= 1
  end
  n
end

def for_loop(items)
  total = 0
  for item in items
    total += item
  end
  total
end

def case_when(v)
  case v
  when 1 then "one"
  when 2 then "two"
  else "many"
  end
end

def begin_rescue(text)
  begin
    Integer(text)
  rescue ArgumentError
    0
  end
end

def boolean_operators(a, b, c)
  a && b || c
end

def word_operators(a, b)
  a and b
end

def ternary(v)
  v > 0 ? 1 : 2
end

def safe_navigation(v)
  v&.name
end

File: tests/fixtures/grammar/constructs.java
// Every control-flow construct in the Java Language Specification that
// can carry a decision. One method per construct.
package grammar;

public class Constructs {
    public int ifElse(int v) {
        if (v > 0) {
            return 1;
        } else if (v < 0) {
            return -1;
        }
        return 0;
    }

    public int forLoop(int n) {
        int total = 0;
        for (int i = 0; i < n; i++) {
            total += i;
        }
        return total;
    }

    public int forEach(int[] items) {
        int total = 0;
        for (int item : items) {
            total += item;
        }
        return total;
    }

    public int whileLoop(int n) {
        while (n > 0) {
            n--;
        }
        return n;
    }

    public int doWhileLoop(int n) {
        do {
            n--;
        } while (n > 0);
        return n;
    }

    public String switchStatement(int v) {
        switch (v) {
            case 1:
                return "one";
            case 2:
                return "two";
            default:
                return "many";
        }
    }

    public int tryCatch(String text) {
        try {
            return Integer.parseInt(text);
        } catch (NumberFormatException e) {
            return 0;
        }
    }

    public boolean booleanOperators(boolean a, boolean b, boolean c) {
        return a && b || c;
    }

    public int ternary(int v) {
        return v > 0 ? 1 : 2;
    }
}
